HyperionScreenCap/Capture/Dx11ScreenCapture.cs

```diff
@@ -234,42 +271,106 @@ private byte[] ManagedCapture()
+                } while( ( mant & 0x400 ) == 0 ); // while not normal
+                mant &= 0x3ff;                    // discard subnormal bit
+            }                                     // else +/-0 -> +/-0
+            return BitConverter.ToSingle(BitConverter.GetBytes(          // combine all parts
+                ( fullFloat & 0x8000 ) << 16          // sign  << ( 31 - 15 )
+                | ( exp | mant ) << 13 ), 0);         // value << ( 23 - 10 )
+        }
         /// <summary>
         /// Reads from the memory locations pointed to by the DataBox and saves it into a byte array
         /// ignoring the alpha component of each pixel.
         /// </summary>
         /// <param name="mapSource"></param>
         /// <returns></returns>
-        private byte[] ToRGBArray(DataBox mapSource)
+        private byte[] ToRGBArray(DataBox mapSource, Format format)
         {
-            var sourcePtr = mapSource.DataPointer;
             byte[] bytes = new byte[CaptureWidth * 3 * CaptureHeight];
             int byteIndex = 0;
-            for ( int y = 0; y < CaptureHeight; y++ )
-            {
-                Int32[] rowData = new Int32[CaptureWidth];
-                Marshal.Copy(sourcePtr, rowData, 0, CaptureWidth);
 
-                foreach ( Int32 pixelData in rowData )
+            if (format == Format.R16G16B16A16_Float)
+            {
+                unsafe
                 {
-                    byte[] values = BitConverter.GetBytes(pixelData);
-                    if ( BitConverter.IsLittleEndian )
+                    byte* ptr = (byte*)mapSource.DataPointer;
+                    byte* rowptr = ptr;
+                    for (int y = 0; y < CaptureHeight; y++)
                     {
-                        // Byte order : bgra
-                        bytes[byteIndex++] = values[2];
-                        bytes[byteIndex++] = values[1];
-                        bytes[byteIndex++] = values[0];
+                        byte* pixelptr = rowptr;
+                        for (int x = 0; x < CaptureWidth; x++)
+                        {
+                            for (int comp = 0; comp < 3; comp++)
+                            {
+                                byte lo = *pixelptr++;
+                                byte hi = *pixelptr++;
+
+                                // No idea why these values range from 4.6 to 0 instead of 1 to 0
+                                // f(x) = sqrt(x/4.6) seems to approximate what the values should be.
+                                bytes[byteIndex++] = (byte)(MathExt.Clamp(Math.Sqrt(Parse16BitFloat(hi, lo) / 4.6), 0, 1) * 255);
+                            }
+                            pixelptr += 2; //skip alpha
+                        }
+                        rowptr += mapSource.RowPitch;
                     }
-                    else
+                }
+            }
+            else if (format == Format.B8G8R8A8_UNorm)
```

Comment:
I think we should add R10G10B10A2_UNorm format: `else if (format == Format.B8G8R8A8_UNorm || format == Format.R10G10B10A2_UNorm)`